pkg/elastic/elasticsearch.go

```diff
@@ -56,211 +54,245 @@ func NewElasticRepository(log *zap.Logger, config *configuration.ElasticsearchCo
 	return repository, nil
 }
 
-func (repository *ElasticRepository) FilterByIndex(index string) ([]string, error) {
-	esClient := repository.esClient
-	searchResult, err := esClient.Search(
-		esClient.Search.WithContext(context.Background()),
-		esClient.Search.WithIndex(index),
-		esClient.Search.WithTrackTotalHits(true),
-		esClient.Search.WithPretty(),
-	)
+func (repository *ElasticRepository) FilterLogs(params logs.Parameters) ([]string, error) {
 
-	var logsList []string // create a slice of type string to append logs to
+	numParameters := 0
+	var queryBuilder []map[string]interface{}
 
-	if err != nil {
-		repository.log.Error("failed exec ES query", zap.Error(err))
-		return logsList, getError(err)
+	if len(params.Namespace) > 0 {
+		term := map[string]interface{}{
+			"term": map[string]interface{}{
+				"kubernetes.namespace_name": params.Namespace},
+		}
+		queryBuilder = append(queryBuilder, term)
+		numParameters = numParameters + 1
 	}
 
-	var result map[string]interface{}
+	if len(params.Podname) > 0 {
 
-	err = json.NewDecoder(searchResult.Body).Decode(&result) //convert searchresult to map[string]interface{}
-	if err != nil {
-		repository.log.Error("Error occurred while decoding JSON", zap.Error(err))
-		return logsList, err
+		term := map[string]interface{}{
+			"term": map[string]interface{}{
+				"kubernetes.pod_name": params.Podname},
+		}
+		queryBuilder = append(queryBuilder, term)
+		numParameters = numParameters + 1
 	}
-
-	if _, ok := result["hits"]; !ok {
-		repository.log.Error("An error occurred while fetching logs", zap.Any("result", result))
-		return logsList, logs.NotFoundError()
+	if len(params.Index) > 0 {
+		term := map[string]interface{}{
+			"term": map[string]interface{}{
+				"_index": fmt.Sprintf("%s", params.Index)},
+		}
+		queryBuilder = append(queryBuilder, term)
+		numParameters = numParameters + 1
+	}
+	if len(params.StartTime) > 0 && len(params.FinishTime) > 0 {
+
+		timeSubquery := map[string]interface{}{
+			"range": map[string]interface{}{
+				"@timestamp": map[string]interface{}{
+					"gte": params.StartTime,
+					"lte": params.FinishTime,
+				},
+			},
+		}
+		queryBuilder = append(queryBuilder, timeSubquery)
+		numParameters = numParameters + 1
+	}
+	if len(params.Level) > 0 {
+		term := map[string]interface{}{
+			"term": map[string]interface{}{
+				"level": params.Level},
+		}
+		queryBuilder = append(queryBuilder, term)
+		numParameters = numParameters + 1
 	}
 
-	logsList = getRelevantLogs(result)
+	query := map[string]interface{}{
+		"query": map[string]interface{}{
+			"bool": map[string]interface{}{
+				"should":               queryBuilder,
+				"minimum_should_match": numParameters}},
+		"size": "1000"}
```

Comment:
@AkshayBhansali18 I assume this is the number of entries to return? This could be a parameter of the query. Something like "max_entries".